src/probe.ts

```diff
@@ -97,26 +98,16 @@ export interface TcpSocketProbeOptions extends ProbeOptions {
  * Probe describes a health check to be performed against a container to
  * determine whether it is alive or ready to receive traffic.
  */
-export abstract class Probe {
```

Comment:
@Chriscbr I refactored this class a bit since I didn't like how we had to explicitly call `parseProbeOptions` 3 times, felt error prone.